### PR TITLE
Feature/smhe 134 smart recipe g meter installation

### PR DESCRIPTION
**Pull request by harrymiddelburgetall**

### CODE REVIEW

File: osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/ConfigurationService.java [hunk 1]
```diff
@@ -672,12 +672,19 @@ public void generateAndReplaceKeys(final MessageMetadata messageMetadata)
     final SmartMeter smartMeter =
         this.domainHelperService.findSmartMeter(messageMetadata.getDeviceIdentification());
 
+    final Device connectToDevice;
+    if (smartMeter.getGatewayDevice() != null) {
+      connectToDevice = smartMeter.getGatewayDevice();
+    } else {
+      connectToDevice = smartMeter;
+    }
```
Comment (stefanermens) [inline osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/ConfigurationService.java:680]: Is this still necessary? We don't use GenerateAndReplaceKeys anymore for the installation of a G meter. Instead we use `SetEncryptionKeyExchangeOnGMeter`. Maybe add an errormessage to indicate that `GenerateAndReplaceKeys` is not supported for mbus meters?

**Comment by kroesctrl** [comment]
https://ci.opensmartgridplatform.org/view/Nightly/job/OSGP_open-smart-grid-platform_nightly/1658/

File: osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/ConfigurationService.java (hunk 1 above)
Comment (kroesctrl) [reply osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/ConfigurationService.java]: I removed this functionality

**Review by kroesctrl — APPROVED** [review]

**Comment by kroesctrl** [comment]
https://ci.opensmartgridplatform.org/view/Nightly/job/OSGP_open-smart-grid-platform_nightly/1661/

**Comment by kroesctrl** [comment]
https://ci.opensmartgridplatform.org/view/Nightly/job/OSGP_open-smart-grid-platform_nightly/1663/

**Review by stefanermens — APPROVED** [review]

**Pull request merged**